code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.kt

```diff
@@ -680,18 +883,60 @@ class EdgeBridgeExtensionTests {
         assertEquals(event.uniqueIdentifier, responseEvent.parentID)
     }
 
+    // Tests that top level properties are unaffected by Analytics keys mapping formatting validation
     @Test
-    fun testHandleTrackEvent_mapsNullAndEmptyValues_andClearsNullValues_dispatchesEdgeRequestEvent() {
+    fun testHandleTrackEvent_mapsNullAndEmptyValues_dispatchesEdgeRequestEvent() {
         val event = Event.Builder("Test Track Event", EventType.GENERIC_TRACK, EventSource.REQUEST_CONTENT)
             .setEventData(
                 mapOf(
                     "key3" to "",
                     "key4" to null,
+                    null to "valueNullKey" // This is filtered out by setEventData
+                )
+            )
+            .build()
+
+        extension.handleTrackRequest(event)
+
+        val responseEvent = captureAndAssertDispatchedEvent()
+
+        val expectedData = mapOf(
+            "data" to mapOf(
+                "key3" to "",
+                "key4" to null
+            ),
+            "xdm" to mapOf(
+                "eventType" to EdgeBridgeTestConstants.JsonValues.EVENT_TYPE,
+                "timestamp" to TimeUtils.getISO8601UTCDateWithMilliseconds(Date(event.timestamp))
+            )
+        )
+
+        assertEquals(expectedData, responseEvent.eventData)
+        assertEquals(event.uniqueIdentifier, responseEvent.parentID)
+    }
+
+    // Test context data values are cleaned such that only String type is allowed.
+    //
+    // NOTE: Android is stricter than iOS because of the Map<String, String> public API requirement
+    // and its enforcement in `cleanContextData` - `char` and number types are not allowed
```

Comment:
I think this test case is fine. Android Analytics is more restrictive as it only allows String types for values, while iOS allows String, Number, and Char. As that is how Analytics has done it in the past, then doing the same for Edge Bridge is fine.